packages/lending/src/adapter.ts

```diff
@@ -0,0 +1,991 @@
+        ? tokenOrTokens.some((token) => !swapper.isSupportedToken(token))
+        : !swapper.isSupportedToken(tokenOrTokens);
+      if (isNotSupported) {
+        continue;
+      }
+      tokensSupportedSwappers.push(swapper);
+
+      bestSwapper = swapper;
+      break;
+    }
+    if (!bestSwapper) {
+      bestSwapper = tokensSupportedSwappers[0] ?? canCustomTokenSwappers[0];
+    }
+    if (!bestSwapper) {
+      bestSwapper = this.swappers[0];
+    }
+
+    return bestSwapper;
+  }
+
+  async getPortfolios(account: string): Promise<Portfolio[]> {
+    const portfolios = await Promise.all(
+      Object.values(this.protocolMap).map((protocol) => protocol.getPortfolios(account))
+    );
+    return flatten(portfolios);
+  }
+
+  async getPortfolio(account: string, protocolId: string, _marketId: string) {
+    return await this.protocolMap[protocolId].getPortfolio(account, _marketId);
+  }
+
+  getProtocol(id: string) {
+    return this.protocolMap[id];
+  }
+
+  // tokenA: USDC, tokenB: USDT
+  // 1. flashloan USDC
+  // 2. swap USDC to USDT
+  // 3. deposit USDT, get aUSDT
+  // 4. return-funds aUSDT to user
+  // 5. add-funds aUSDC to router
+  // 6. withdraw aUSDC, get USDC
+  // 7. flashloan repay USDC
+  async getCollateralSwap(
+    protocolId: string,
+    marketId: string,
+    params: BaseParams,
+    account: string,
+    portfolio?: Portfolio
+  ): Promise<BaseFields> {
+    const { srcAmount } = params;
+    const srcToken = common.classifying(params.srcToken);
+    const destToken = common.classifying(params.destToken);
+    const wrappedSrcToken = srcToken.wrapped;
+    const wrappedDestToken = destToken.wrapped;
+
+    const collateralSwapLogics: apisdk.Logic<any>[] = [];
+    const protocol = this.getProtocol(protocolId);
+
+    portfolio = portfolio || (await protocol.getPortfolio(account, marketId));
+    const afterPortfolio = portfolio.clone();
+
+    // compound v3 base token supplied not support collateral swap
+    if (protocolId === 'compound-v3' && portfolio.baseToken?.is(srcToken.unwrapped))
+      throw new Error('Compound V3 does not support the base token for performing collateral swaps');
+
+    // ---------- flashloan ----------
+    const flashLoanAggregatorQuotation = await apisdk.protocols.utility.getFlashLoanAggregatorQuotation(this.chainId, {
+      repays: [{ token: wrappedSrcToken, amount: srcAmount }],
+    });
+
+    const flashLoanTokenAmount = flashLoanAggregatorQuotation.loans.tokenAmountMap[wrappedSrcToken.address];
+
+    const [flashLoanLoanLogic, flashLoanRepayLogic] = apisdk.protocols.utility.newFlashLoanAggregatorLogicPair(
+      flashLoanAggregatorQuotation.protocolId,
+      flashLoanAggregatorQuotation.loans.toArray()
+    );
+    collateralSwapLogics.push(flashLoanLoanLogic);
+
+    // ---------- swap ----------
+    const swapper = this.findSwapper([wrappedSrcToken, wrappedDestToken]);
+    const swapQuotation = await swapper.quote({
+      input: flashLoanTokenAmount,
```

Comment:
suggest using srcAmount

Reply:
Why suggest using srcAmount? Isn't the swapping do the swap for the entire flashLoan amount?

Reply:
Yes. I was thinking that it's better to use all the variables in params (eg. srcToken, srcAmount) for reviewers to read the logic easily (it would be hard to check the value of flashLoanTokenAmount). But it's also right that swap amount comes from flashloan amount which comes from srcAmount. I guess this is a decision to make to see if we want to follow the flow of logic or to use limited variables for better understanding.